core/primitives/src/account.rs

```diff
@@ -24,6 +28,45 @@ impl Account {
     }
 }
 
+/// Calculates the storage and the name rent for the given account
+/// for period of account.storage_paid_at to current block_height
+pub fn calculate_rent(
+    account_id: &AccountId,
+    account: &Account,
+    // TODO #1903 block_height: BlockHeight,
+    block_index: BlockHeight,
+    account_length_baseline_cost_per_block: Balance,
+    storage_cost_byte_per_block: Balance,
+) -> Balance {
+    let charge = u128::from(block_index - account.storage_paid_at)
```

Comment:
This can underflow if the account state is invalid. I don't know if we need to care about this during the RPC request, but this function might be used somewhere outside of the RPCs scope and it may operate on the invalid state, so it will panic the node.